In [1]:
# This is file for train, prediction

import pandas as pd

dirpath = 'C:/Users/rihot/Desktop/Deep_learning/DACON_used_car_price/'

train = pd.read_csv('data/preprocessed_train.csv')
test = pd.read_csv('data/preprocessed_test.csv')

In [2]:
Y = train[['target']].values
X = train[ ['title', 'odometer', 'location', 'isimported', 'engine', 'paint', 'year', 'brand' ] ].values

X_test = test[ ['title', 'odometer', 'location', 'isimported', 'engine', 'paint', 'year', 'brand' ] ].values

# X = train[ ['title', 'odometer', 'location', 'isimported', 'engine', 'transmission', 'fuel', 'paint', 'year', 'brand' ] ].values
#
# X_test = test[ ['title', 'odometer', 'location', 'isimported', 'engine', 'transmission', 'fuel', 'paint', 'year', 'brand' ] ].values

In [3]:
X.shape, Y.shape

((1015, 8), (1015, 1))

In [4]:
from sklearn.preprocessing import MinMaxScaler

scalerX = MinMaxScaler()
scalerX.fit(X)
X = scalerX.transform(X)
X_test = scalerX.transform(X_test)

scalerY = MinMaxScaler()
scalerY.fit(Y)
Y = scalerY.transform(Y)

In [5]:
from sklearn.metrics import mean_squared_error
from sklearn.model_selection import train_test_split
import optuna
from optuna import Trial
from lightgbm import LGBMRegressor
import xgboost as xgb

def LGBM(trial : Trial, X, Y, test):
    param = {
        'objective': 'regression', # 회귀
        'metric': 'rmse',
        "n_estimators" : trial.suggest_categorical('n_estimators', [500, 600,700,800,900,1000,1200,1400,1600,1800,2000, 4000]),
        'max_depth':trial.suggest_int('max_depth', 8, 16),
        'learning_rate': trial.suggest_categorical('learning_rate', [0.01,0.02,0.05]),
        # 'nthread' : -1,
        'subsample': trial.suggest_categorical('subsample', [0.6,0.7,0.8,1.0] ),
        'random_state': 42
    }
    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)

    y_train = y_train.reshape(-1, 1)
    y_test  = y_test.reshape(-1, 1)

    model = LGBMRegressor(**param)
    LGBM_model = model.fit(X_train, y_train, verbose=False, eval_set=[(X_test, y_test)])
    score = mean_squared_error(LGBM_model.predict(X_test), y_test, squared=False)

    return score

In [6]:
import warnings
warnings.filterwarnings('ignore')

In [7]:
import numpy as np
from optuna.samplers import TPESampler

study = optuna.create_study(direction='minimize',sampler=TPESampler())
study.optimize(lambda trial : LGBM(trial, X,  Y, X_test), n_trials=50)
print('Best trial: score {},\nparams {}'.format(study.best_trial.value,study.best_trial.params))

[I 2022-06-02 23:55:31,269] A new study created in memory with name: no-name-c5b49c00-f899-479e-a6e2-1049aab8bae2
[I 2022-06-02 23:55:33,461] Trial 0 finished with value: 0.05465324373714255 and parameters: {'n_estimators': 4000, 'max_depth': 14, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 0 with value: 0.05465324373714255.
[I 2022-06-02 23:55:33,982] Trial 1 finished with value: 0.034577816643479066 and parameters: {'n_estimators': 1400, 'max_depth': 8, 'learning_rate': 0.02, 'subsample': 0.7}. Best is trial 1 with value: 0.034577816643479066.
[I 2022-06-02 23:55:34,260] Trial 2 finished with value: 0.04663499004303781 and parameters: {'n_estimators': 600, 'max_depth': 9, 'learning_rate': 0.01, 'subsample': 0.6}. Best is trial 1 with value: 0.034577816643479066.
[I 2022-06-02 23:55:34,671] Trial 3 finished with value: 0.04226636350908355 and parameters: {'n_estimators': 700, 'max_depth': 13, 'learning_rate': 0.05, 'subsample': 0.6}. Best is trial 1 with value: 0.0345778166

Best trial: score 0.02670091692246342,
params {'n_estimators': 2000, 'max_depth': 9, 'learning_rate': 0.02, 'subsample': 0.8}


Best trial: score 0.02950600044760606,
params {'n_estimators': 500, 'max_depth': 12, 'learning_rate': 0.02, 'subsample': 0.8}

In [9]:
from xgboost import XGBRegressor

def XGB(trial : Trial, X, Y, test):
    param = {
        # 'objective': 'regression', # 회귀
        'metric': 'rmse',
        "n_estimators" : trial.suggest_categorical('n_estimators', [500, 600,700,800,900,1000,1200,1400,1600,1800,2000, 4000]),
        'max_depth':trial.suggest_int('max_depth', 8, 16),
        'learning_rate': trial.suggest_categorical('learning_rate', [0.01,0.02,0.05]),
        # 'nthread' : -1,
        'subsample': trial.suggest_categorical('subsample', [0.6,0.7,0.8,1.0] ),
        'random_state': 42
    }
    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)

    y_train = y_train.reshape(-1, 1)
    y_test  = y_test.reshape(-1, 1)

    model = XGBRegressor(**param)
    XGB_model = model.fit(X_train, y_train, verbose=False, eval_set=[(X_test, y_test)])
    score = mean_squared_error(XGB_model.predict(X_test), y_test, squared=False)

    return score

study = optuna.create_study(direction='minimize',sampler=TPESampler())
study.optimize(lambda trial : XGB(trial, X,  Y, X_test), n_trials=50)

print('Best trial: score {},\nparams {}'.format(study.best_trial.value,study.best_trial.params))

[I 2022-06-03 00:01:19,042] A new study created in memory with name: no-name-333f7c87-1662-4968-b267-896f708fb965


[00:01:19] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:19,926] Trial 0 finished with value: 0.0839424345688986 and parameters: {'n_estimators': 1800, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.6}. Best is trial 0 with value: 0.0839424345688986.


[00:01:19] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:20,592] Trial 1 finished with value: 0.05331714185330453 and parameters: {'n_estimators': 800, 'max_depth': 15, 'learning_rate': 0.01, 'subsample': 0.6}. Best is trial 1 with value: 0.05331714185330453.


[00:01:20] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:21,111] Trial 2 finished with value: 0.03864030830845845 and parameters: {'n_estimators': 800, 'max_depth': 14, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 2 with value: 0.03864030830845845.


[00:01:21] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:22,340] Trial 3 finished with value: 0.0314801292227182 and parameters: {'n_estimators': 4000, 'max_depth': 11, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 3 with value: 0.0314801292227182.


[00:01:22] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:22,925] Trial 4 finished with value: 0.02088880864021991 and parameters: {'n_estimators': 900, 'max_depth': 16, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:22] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:23,411] Trial 5 finished with value: 0.0266136234609935 and parameters: {'n_estimators': 700, 'max_depth': 10, 'learning_rate': 0.01, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:23] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:24,485] Trial 6 finished with value: 0.06562168514879387 and parameters: {'n_estimators': 2000, 'max_depth': 11, 'learning_rate': 0.02, 'subsample': 0.6}. Best is trial 4 with value: 0.02088880864021991.


[00:01:24] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:24,955] Trial 7 finished with value: 0.03884459563319258 and parameters: {'n_estimators': 600, 'max_depth': 13, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:24] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:25,281] Trial 8 finished with value: 0.07681765616063248 and parameters: {'n_estimators': 500, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:25] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:26,216] Trial 9 finished with value: 0.04982225553968342 and parameters: {'n_estimators': 1800, 'max_depth': 9, 'learning_rate': 0.02, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:26] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:26,953] Trial 10 finished with value: 0.07420236285360123 and parameters: {'n_estimators': 900, 'max_depth': 16, 'learning_rate': 0.02, 'subsample': 0.8}. Best is trial 4 with value: 0.02088880864021991.


[00:01:26] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:27,935] Trial 11 finished with value: 0.03316651225149414 and parameters: {'n_estimators': 1400, 'max_depth': 10, 'learning_rate': 0.01, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:27] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:28,580] Trial 12 finished with value: 0.033209433583675424 and parameters: {'n_estimators': 700, 'max_depth': 12, 'learning_rate': 0.01, 'subsample': 0.8}. Best is trial 4 with value: 0.02088880864021991.


[00:01:28] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:29,532] Trial 13 finished with value: 0.05985264000296058 and parameters: {'n_estimators': 1000, 'max_depth': 16, 'learning_rate': 0.01, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:29] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:30,319] Trial 14 finished with value: 0.045339089192124836 and parameters: {'n_estimators': 900, 'max_depth': 13, 'learning_rate': 0.01, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:30] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:30,802] Trial 15 finished with value: 0.07492602213209931 and parameters: {'n_estimators': 700, 'max_depth': 10, 'learning_rate': 0.01, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:30] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:31,739] Trial 16 finished with value: 0.044135861272407415 and parameters: {'n_estimators': 1600, 'max_depth': 14, 'learning_rate': 0.02, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:31] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:32,728] Trial 17 finished with value: 0.04899212511878239 and parameters: {'n_estimators': 1200, 'max_depth': 12, 'learning_rate': 0.01, 'subsample': 0.8}. Best is trial 4 with value: 0.02088880864021991.


[00:01:32] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:33,276] Trial 18 finished with value: 0.060319887102622566 and parameters: {'n_estimators': 900, 'max_depth': 10, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:33] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:33,731] Trial 19 finished with value: 0.030072675670233397 and parameters: {'n_estimators': 700, 'max_depth': 15, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:33] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:34,929] Trial 20 finished with value: 0.03674349760273552 and parameters: {'n_estimators': 2000, 'max_depth': 9, 'learning_rate': 0.01, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:34] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:35,358] Trial 21 finished with value: 0.027768862327682567 and parameters: {'n_estimators': 700, 'max_depth': 15, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:35] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:35,770] Trial 22 finished with value: 0.07647510228750672 and parameters: {'n_estimators': 700, 'max_depth': 15, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:35] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:36,204] Trial 23 finished with value: 0.05189288186905545 and parameters: {'n_estimators': 700, 'max_depth': 16, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:36] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:36,568] Trial 24 finished with value: 0.05086640073734041 and parameters: {'n_estimators': 500, 'max_depth': 14, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:36] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:38,184] Trial 25 finished with value: 0.03942395903486626 and parameters: {'n_estimators': 4000, 'max_depth': 15, 'learning_rate': 0.02, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:38] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:38,755] Trial 26 finished with value: 0.05102813028645873 and parameters: {'n_estimators': 1400, 'max_depth': 13, 'learning_rate': 0.05, 'subsample': 0.8}. Best is trial 4 with value: 0.02088880864021991.


[00:01:38] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:39,705] Trial 27 finished with value: 0.040082603123142664 and parameters: {'n_estimators': 1200, 'max_depth': 16, 'learning_rate': 0.02, 'subsample': 0.6}. Best is trial 4 with value: 0.02088880864021991.


[00:01:39] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:40,515] Trial 28 finished with value: 0.04071870511497466 and parameters: {'n_estimators': 1000, 'max_depth': 12, 'learning_rate': 0.01, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:40] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:41,190] Trial 29 finished with value: 0.03817250018520647 and parameters: {'n_estimators': 1600, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.6}. Best is trial 4 with value: 0.02088880864021991.


[00:01:41] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:41,465] Trial 30 finished with value: 0.03310055662224667 and parameters: {'n_estimators': 600, 'max_depth': 11, 'learning_rate': 0.05, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:41] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:41,901] Trial 31 finished with value: 0.03181409668849503 and parameters: {'n_estimators': 700, 'max_depth': 15, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:41] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:42,410] Trial 32 finished with value: 0.05259772691120984 and parameters: {'n_estimators': 700, 'max_depth': 15, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:42] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:42,817] Trial 33 finished with value: 0.08044739892979842 and parameters: {'n_estimators': 700, 'max_depth': 14, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:42] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:43,252] Trial 34 finished with value: 0.041297829872536586 and parameters: {'n_estimators': 800, 'max_depth': 15, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:43] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:43,778] Trial 35 finished with value: 0.03259806710298513 and parameters: {'n_estimators': 900, 'max_depth': 16, 'learning_rate': 0.05, 'subsample': 0.6}. Best is trial 4 with value: 0.02088880864021991.


[00:01:43] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:44,769] Trial 36 finished with value: 0.04833135362564951 and parameters: {'n_estimators': 1800, 'max_depth': 14, 'learning_rate': 0.02, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:44] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:45,355] Trial 37 finished with value: 0.044151484610059764 and parameters: {'n_estimators': 700, 'max_depth': 13, 'learning_rate': 0.01, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:45] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:46,621] Trial 38 finished with value: 0.03544117618939772 and parameters: {'n_estimators': 4000, 'max_depth': 16, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:46] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:47,096] Trial 39 finished with value: 0.03831009934338931 and parameters: {'n_estimators': 700, 'max_depth': 9, 'learning_rate': 0.02, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:47] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:47,817] Trial 40 finished with value: 0.0432718084827138 and parameters: {'n_estimators': 800, 'max_depth': 15, 'learning_rate': 0.02, 'subsample': 0.6}. Best is trial 4 with value: 0.02088880864021991.


[00:01:47] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:49,072] Trial 41 finished with value: 0.04661773370846398 and parameters: {'n_estimators': 4000, 'max_depth': 11, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:49] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:50,359] Trial 42 finished with value: 0.07468150384416392 and parameters: {'n_estimators': 4000, 'max_depth': 11, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:50] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:51,108] Trial 43 finished with value: 0.05599024696707068 and parameters: {'n_estimators': 2000, 'max_depth': 10, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:51] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:51,446] Trial 44 finished with value: 0.08200606342807897 and parameters: {'n_estimators': 600, 'max_depth': 11, 'learning_rate': 0.05, 'subsample': 0.8}. Best is trial 4 with value: 0.02088880864021991.


[00:01:51] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:52,202] Trial 45 finished with value: 0.035065988611341234 and parameters: {'n_estimators': 900, 'max_depth': 12, 'learning_rate': 0.01, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:52] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:52,537] Trial 46 finished with value: 0.04616227808928924 and parameters: {'n_estimators': 500, 'max_depth': 9, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:52] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:53,955] Trial 47 finished with value: 0.06490778869498891 and parameters: {'n_estimators': 1800, 'max_depth': 16, 'learning_rate': 0.01, 'subsample': 0.7}. Best is trial 4 with value: 0.02088880864021991.


[00:01:53] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:54,608] Trial 48 finished with value: 0.0492064880653608 and parameters: {'n_estimators': 1400, 'max_depth': 10, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 4 with value: 0.02088880864021991.


[00:01:54] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.6.0/src/learner.cc:627: 
Parameters: { "metric" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




[I 2022-06-03 00:01:55,180] Trial 49 finished with value: 0.08382543566422782 and parameters: {'n_estimators': 700, 'max_depth': 13, 'learning_rate': 0.01, 'subsample': 0.8}. Best is trial 4 with value: 0.02088880864021991.


Best trial: score 0.02088880864021991,
params {'n_estimators': 900, 'max_depth': 16, 'learning_rate': 0.02, 'subsample': 1.0}


Best trial: score 0.02088880864021991,
params {'n_estimators': 900, 'max_depth': 16, 'learning_rate': 0.02, 'subsample': 1.0}

In [12]:
from catboost import CatBoostRegressor

def cat(trial : Trial, X, Y, test):
    param = {
        # 'objective': 'regression', # 회귀
        # 'metric': 'rmse',
        "n_estimators" : trial.suggest_categorical('n_estimators', [500, 600,700,800,900,1000,1200,1400,1600,1800,2000, 4000]),
        'max_depth':trial.suggest_int('max_depth', 8, 16),
        'learning_rate': trial.suggest_categorical('learning_rate', [0.01,0.02,0.05]),
        # 'nthread' : -1,
        'subsample': trial.suggest_categorical('subsample', [0.6,0.7,0.8,1.0] ),
        'random_state': 42
    }
    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)

    y_train = y_train.reshape(-1, 1)
    y_test  = y_test.reshape(-1, 1)

    model = CatBoostRegressor(**param)
    cat_model = model.fit(X_train, y_train, verbose=False, eval_set=[(X_test, y_test)])
    score = mean_squared_error(cat_model.predict(X_test), y_test, squared=False)

    return score

study = optuna.create_study(direction='minimize',sampler=TPESampler())
study.optimize(lambda trial : cat(trial, X,  Y, X_test), n_trials=50)
print('Best trial: score {},\nparams {}'.format(study.best_trial.value,study.best_trial.params))

[I 2022-06-03 00:03:17,052] A new study created in memory with name: no-name-7f36e082-a840-4fc4-bc43-d795703688ff
[I 2022-06-03 00:03:56,946] Trial 0 finished with value: 0.06047475812189175 and parameters: {'n_estimators': 1400, 'max_depth': 12, 'learning_rate': 0.02, 'subsample': 0.6}. Best is trial 0 with value: 0.06047475812189175.
[I 2022-06-03 00:04:03,142] Trial 1 finished with value: 0.05307147628981088 and parameters: {'n_estimators': 800, 'max_depth': 10, 'learning_rate': 0.02, 'subsample': 1.0}. Best is trial 1 with value: 0.05307147628981088.
[I 2022-06-03 00:04:20,953] Trial 2 finished with value: 0.044453107955052716 and parameters: {'n_estimators': 1200, 'max_depth': 11, 'learning_rate': 0.02, 'subsample': 0.8}. Best is trial 2 with value: 0.044453107955052716.
[I 2022-06-03 00:04:24,641] Trial 3 finished with value: 0.04991663524519707 and parameters: {'n_estimators': 1600, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.6}. Best is trial 2 with value: 0.044453107

Best trial: score 0.02228076617330138,
params {'n_estimators': 800, 'max_depth': 10, 'learning_rate': 0.02, 'subsample': 0.8}


Best trial: score 0.02228076617330138,
params {'n_estimators': 800, 'max_depth': 10, 'learning_rate': 0.02, 'subsample': 0.8}

In [19]:
from sklearn.ensemble import GradientBoostingRegressor

def gradi(trial : Trial, X, Y, test):
    param = {
        # 'objective': 'regression', # 회귀
        # 'metric': 'rmse',
        "n_estimators" : trial.suggest_categorical('n_estimators', [500, 600,700,800,900,1000,1200,1400,1600,1800,2000, 4000]),
        'max_depth':trial.suggest_int('max_depth', 8, 16),
        'learning_rate': trial.suggest_categorical('learning_rate', [0.01,0.02,0.05]),
        # 'nthread' : -1,
        'subsample': trial.suggest_categorical('subsample', [0.6,0.7,0.8,1.0] ),
        'random_state': 42
    }
    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)

    y_train = y_train.reshape(-1, 1)
    y_test  = y_test.reshape(-1, 1)

    model = GradientBoostingRegressor(**param)
    gradi_model = model.fit(X_train, y_train)
    score = mean_squared_error(gradi_model.predict(X_test), y_test, squared=False)

    return score

study = optuna.create_study(direction='minimize',sampler=TPESampler())
study.optimize(lambda trial : gradi(trial, X,  Y, X_test), n_trials=50)
print('Best trial: score {},\nparams {}'.format(study.best_trial.value,study.best_trial.params))

[I 2022-06-03 01:55:12,903] A new study created in memory with name: no-name-618a3711-610d-442b-a1d2-24f39ab1fea4
[I 2022-06-03 01:55:15,419] Trial 0 finished with value: 0.05914506379597146 and parameters: {'n_estimators': 1200, 'max_depth': 12, 'learning_rate': 0.02, 'subsample': 0.8}. Best is trial 0 with value: 0.05914506379597146.
[I 2022-06-03 01:55:16,592] Trial 1 finished with value: 0.03949706502543835 and parameters: {'n_estimators': 900, 'max_depth': 9, 'learning_rate': 0.02, 'subsample': 0.6}. Best is trial 1 with value: 0.03949706502543835.
[I 2022-06-03 01:55:19,545] Trial 2 finished with value: 0.05288881995786198 and parameters: {'n_estimators': 1400, 'max_depth': 11, 'learning_rate': 0.05, 'subsample': 1.0}. Best is trial 1 with value: 0.03949706502543835.
[I 2022-06-03 01:55:21,107] Trial 3 finished with value: 0.08332525123170709 and parameters: {'n_estimators': 800, 'max_depth': 13, 'learning_rate': 0.05, 'subsample': 0.7}. Best is trial 1 with value: 0.039497065025

Best trial: score 0.02844488766002739,
params {'n_estimators': 700, 'max_depth': 9, 'learning_rate': 0.05, 'subsample': 0.8}


Best trial: score 0.02844488766002739,
params {'n_estimators': 700, 'max_depth': 9, 'learning_rate': 0.05, 'subsample': 0.8}

In [26]:
from sklearn.ensemble import ExtraTreesRegressor

def etr(trial : Trial, X, Y, test):
    param = {
        # 'objective': 'regression', # 회귀
        # 'metric': 'rmse',
        "n_estimators" : trial.suggest_categorical('n_estimators', [500, 600,700,800,900,1000,1200,1400,1600,1800,2000, 4000]),
        'max_depth':trial.suggest_int('max_depth', 8, 16),
        # 'learning_rate': trial.suggest_categorical('learning_rate', [0.01,0.02,0.05]),
        # 'nthread' : -1,
        # 'subsample': trial.suggest_categorical('subsample', [0.6,0.7,0.8,1.0] ),
        'random_state': 42
    }
    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)

    y_train = y_train.reshape(-1, 1)
    y_test  = y_test.reshape(-1, 1)

    model = ExtraTreesRegressor(**param)
    etr_model = model.fit(X_train, y_train)
    score = mean_squared_error(etr_model.predict(X_test), y_test, squared=False)

    return score

study = optuna.create_study(direction='minimize',sampler=TPESampler())
study.optimize(lambda trial : etr(trial, X,  Y, X_test), n_trials=50)
print('Best trial: score {},\nparams {}'.format(study.best_trial.value,study.best_trial.params))

[I 2022-06-03 01:58:00,537] A new study created in memory with name: no-name-dd0b8811-9325-4723-8262-e0e3fe5213ca
[I 2022-06-03 01:58:08,097] Trial 0 finished with value: 0.09096298022115129 and parameters: {'n_estimators': 4000, 'max_depth': 15}. Best is trial 0 with value: 0.09096298022115129.
[I 2022-06-03 01:58:09,217] Trial 1 finished with value: 0.02877347975814291 and parameters: {'n_estimators': 600, 'max_depth': 15}. Best is trial 1 with value: 0.02877347975814291.
[I 2022-06-03 01:58:15,637] Trial 2 finished with value: 0.051236962110817054 and parameters: {'n_estimators': 4000, 'max_depth': 13}. Best is trial 1 with value: 0.02877347975814291.
[I 2022-06-03 01:58:17,146] Trial 3 finished with value: 0.045068519397612514 and parameters: {'n_estimators': 700, 'max_depth': 15}. Best is trial 1 with value: 0.02877347975814291.
[I 2022-06-03 01:58:18,451] Trial 4 finished with value: 0.05819010410208789 and parameters: {'n_estimators': 700, 'max_depth': 13}. Best is trial 1 with 

Best trial: score 0.02668045501392634,
params {'n_estimators': 600, 'max_depth': 9}


Best trial: score 0.02668045501392634,
params {'n_estimators': 600, 'max_depth': 9}

In [32]:
from sklearn.ensemble import RandomForestRegressor

def rfr(trial : Trial, X, Y, test):
    param = {
        # 'objective': 'regression', # 회귀
        # 'metric': 'rmse',
        "n_estimators" : trial.suggest_categorical('n_estimators', [500, 600,700,800,900,1000,1200,1400,1600,1800,2000, 4000]),
        'max_depth':trial.suggest_int('max_depth', 8, 16),
        # 'learning_rate': trial.suggest_categorical('learning_rate', [0.01,0.02,0.05]),
        # 'nthread' : -1,
        # 'subsample': trial.suggest_categorical('subsample', [0.6,0.7,0.8,1.0] ),
        'random_state': 42
    }
    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)

    y_train = y_train.reshape(-1, 1)
    y_test  = y_test.reshape(-1, 1)

    model = RandomForestRegressor(**param)
    rfr_model = model.fit(X_train, y_train)
    score = mean_squared_error(rfr_model.predict(X_test), y_test, squared=False)

    return score

study = optuna.create_study(direction='minimize',sampler=TPESampler())
study.optimize(lambda trial : rfr(trial, X,  Y, X_test), n_trials=50)
print('Best trial: score {},\nparams {}'.format(study.best_trial.value,study.best_trial.params))

[I 2022-06-03 02:15:57,377] A new study created in memory with name: no-name-8870f84b-4218-4298-b11d-01d21a3ace53
[I 2022-06-03 02:15:58,548] Trial 0 finished with value: 0.04131496640430345 and parameters: {'n_estimators': 600, 'max_depth': 8}. Best is trial 0 with value: 0.04131496640430345.
[I 2022-06-03 02:16:08,172] Trial 1 finished with value: 0.03484342822735787 and parameters: {'n_estimators': 4000, 'max_depth': 12}. Best is trial 1 with value: 0.03484342822735787.
[I 2022-06-03 02:16:09,348] Trial 2 finished with value: 0.0805269800774792 and parameters: {'n_estimators': 500, 'max_depth': 12}. Best is trial 1 with value: 0.03484342822735787.
[I 2022-06-03 02:16:13,447] Trial 3 finished with value: 0.0884373574909707 and parameters: {'n_estimators': 1600, 'max_depth': 14}. Best is trial 1 with value: 0.03484342822735787.
[I 2022-06-03 02:16:15,792] Trial 4 finished with value: 0.03472087963963781 and parameters: {'n_estimators': 900, 'max_depth': 15}. Best is trial 4 with value

Best trial: score 0.033730265932558116,
params {'n_estimators': 2000, 'max_depth': 10}


Best trial: score 0.033730265932558116,
params {'n_estimators': 2000, 'max_depth': 10}